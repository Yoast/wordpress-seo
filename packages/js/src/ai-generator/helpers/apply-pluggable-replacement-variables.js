import { get } from "lodash";
import { languageProcessing } from "yoastseo";
import { EDIT_TYPE } from "../constants";


/**
 * @param {string} content The content.
 * @param {string} [editType] The edit type. See EDIT_TYPE.
 * @returns {string} The content with replaced variables.
 */
export const applyPluggableReplacementVariables = ( content, editType = EDIT_TYPE.title ) => {
	const applyReplaceUsingPlugin = get(
		window,
		"yoast.editorModules.helpers.replacementVariableHelpers.applyReplaceUsingPlugin",
		( data ) => ( {
			url: data.url,
			title: languageProcessing.stripHTMLTags( data.title ),
			description: languageProcessing.stripHTMLTags( data.description ),
		} )
	);
	const CleanContent = languageProcessing.stripSpaces( content );
	const replaced = applyReplaceUsingPlugin( {
		// Ensure the types are all present. This prevents an error as "replace" is always executed on them.
		title: "",
		description: "",
		// Override with the wanted type's content.
		[ editType ]: CleanContent,
	} );
	return get( replaced, editType, content );
};
