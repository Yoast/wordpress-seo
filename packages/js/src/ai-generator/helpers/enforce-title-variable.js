import { TITLE_VARIABLE_REPLACE } from "../constants";

/**
 * Enforces the title replacement variable ('%%title%%' for posts, '%%term_title%%' for terms) in a template.
 *
 * @param {string} template The template.
 * @param {string} contentType The content type ('post' or 'term').
 * @returns {string} The template; either including the replacement variable or being the replacement variable.
 */
export const enforceTitleVariable = ( template, contentType ) =>
	template.includes( TITLE_VARIABLE_REPLACE[ contentType ] ) ? template : TITLE_VARIABLE_REPLACE[ contentType ];
