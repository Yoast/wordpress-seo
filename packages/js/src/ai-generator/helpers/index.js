export { applyPluggableReplacementVariables } from "./apply-pluggable-replacement-variables";
export { enforceTitleVariable } from "./enforce-title-variable";
export { fetchSuggestions, removesLocaleVariantSuffixes } from "./fetch-suggestions";
export { focusFocusKeyphraseInput } from "./focus-focus-keyphrase-input";
export { isConsideredEmpty } from "./is-considered-empty";
export { preparePromptContent } from "./prepare-prompt-content";
