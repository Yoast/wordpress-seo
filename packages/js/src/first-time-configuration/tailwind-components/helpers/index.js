import { __ } from "@wordpress/i18n";
import { cloneDeep, pickBy } from "lodash";

import classNames from "classnames";

export const socialMedia = [
	{ name: "Facebook", placeholder: __( "E.g. https://www.facebook.com/yoast", "wordpress-seo" ) },
	{ name: "Instagram", placeholder: __( "E.g. https://www.instagram.com/yoast", "wordpress-seo" ) },
	{ name: "LinkedIn", placeholder: __( "E.g. https://www.linkedin.com/yoast", "wordpress-seo" ) },
	{ name: "MySpace", placeholder: __( "E.g. https://www.myspace.com/yoast", "wordpress-seo" ) },
	{ name: "Pinterest", placeholder: __( "E.g. https://www.pinterest.com/yoast", "wordpress-seo" ) },
	{ name: "SoundCloud", placeholder: __( "E.g. https://www.soundcloud.com/yoast", "wordpress-seo" ) },
	{ name: "Tumblr", placeholder: __( "E.g. https://www.tumblr.com/yoast", "wordpress-seo" ) },
	{ name: "Twitter", placeholder: __( "E.g. https://www.twitter.com/yoast", "wordpress-seo" ) },
	{ name: "YouTube", placeholder: __( "E.g. https://www.youtube.com/yoast", "wordpress-seo" ) },
	{ name: "Wikipedia", placeholder: __( "E.g. https://www.wikipedia.com/yoast", "wordpress-seo" ) },
];

/**
 * Creates the error ID for the error component.
 *
 * @param {string} inputId The id of the input component.
 *
 * @returns {string} The ID for the error component.
 */
export const getErrorId = ( inputId ) => `error-${ inputId }`;

/**
  * Get props needed to properly display an error in input components.
  *
  * @param {string} inputId The id of the input component.
  * @param {ValidationError} error The error object.
  * @param {boolean} error.isVisible The error object.
  *
  * @returns {Object} Object containing relevant props for displaying.
  */
export const getErrorAriaProps = ( inputId, { isVisible } ) => isVisible ? {
	"aria-invalid": true,
	"aria-describedby": getErrorId( inputId ),
} : {};

/**
 * Helper function to get active styles for select options.
 *
 * @param {boolean} options.active Whether the option is active.
 *
 * @returns {string} Styles for an active option.
 */
export function getOptionActiveStyles( { active, selected } ) {
	return classNames(
		"yst-relative yst-cursor-default yst-select-none yst-py-2 yst-pl-3 yst-pr-9 yst-my-0",
		selected && "yst-bg-primary-500 yst-text-white",
		( active && ! selected ) && "yst-bg-primary-200 yst-text-slate-700",
		( ! active && ! selected ) && "yst-text-slate-700"
	);
}


/**
 * Adds a step to editedSteps if not there already.
 *
 * @param {Array} editedSteps Steps that have been edited.
 * @param {number} stepNumber  The number of the field that was edited.
 *
 * @returns {Array} The new array of editedSteps.
 */
function addStepToEditedSteps( editedSteps, stepNumber ) {
	if ( editedSteps.includes( stepNumber ) ) {
		return [ ...editedSteps ];
	}
	return [ ...editedSteps, stepNumber ];
}

/**
 * Adjusts the editedSteps and returns the full state;
 *
 * @param {Object} state      The state.
 * @param {number} stepNumber The number of the step that was edited.
 *
 * @returns {Object} The new state;
 */
function handleStepEdit( state, stepNumber ) {
	const newEditedSteps = addStepToEditedSteps( state.editedSteps, stepNumber );
	return {
		...state,
		editedSteps: newEditedSteps,
	};
}

/**
 * Manages updating of the error notices when one (or more) field are removed from the user
 *
 * @param {object} prevState An array of fields names with errors
 * @param {int} index The index of the field to be removed
 *
 * @returns {Array} The filtered list of fields with errors
 */
export function removeFieldAndUpdateErrors( prevState, index ) {
	// Map through all the fields with errors
	return prevState.map( ( errorField ) => {
		// Find the index of the current error
		const errorFieldIndex = parseInt( errorField.replace( "other_social_urls-", "" ), 10 );
		// If the error occurs on the field to be removed, mark it for removal
		if ( errorFieldIndex === index ) {
			return "remove";
		// If the field index is past the field to be removed, decrease it by one since we'll remove the previous field
		} else if ( errorFieldIndex > index ) {
			return `other_social_urls-${ errorFieldIndex - 1 }`;
		}
		return errorField;
	// Keep all the errors not marked to be removed
	} ).filter( errorField => errorField !== "remove" );
}

/* eslint-disable complexity */
/**
 * A reducer for the configuration's internal state.
 *
 * @param {Object} state  The "current" state.
 * @param {Object} action The action with which to mutate the state.
 *
 * @returns {Object} The state as altered by the action.
 */
export function configurationReducer( state, action ) {
	let newState = cloneDeep( state );
	switch ( action.type ) {
		case "SET_COMPANY_OR_PERSON":
			newState = handleStepEdit( newState, 2 );
			newState.companyOrPerson = action.payload;
			newState.companyOrPersonLabel = newState.companyOrPersonOptions.filter( ( item ) => {
				return item.value === action.payload;
			} ).pop().label;
			return newState;
		case "CHANGE_COMPANY_NAME":
			newState = handleStepEdit( newState, 2 );
			newState.companyName = action.payload;
			return newState;
		case "SET_COMPANY_LOGO":
			newState = handleStepEdit( newState, 2 );
			newState.companyLogo = action.payload.url;
			newState.companyLogoId = action.payload.id;
			return newState;
		case "REMOVE_COMPANY_LOGO":
			newState = handleStepEdit( newState, 2 );
			newState.companyLogo = "";
			newState.companyLogoId = "";
			return newState;
		case "CHANGE_WEBSITE_NAME":
			newState = handleStepEdit( newState, 2 );
			newState.websiteName = action.payload;
			return newState;
		case "SET_PERSON_LOGO":
			newState = handleStepEdit( newState, 2 );
			newState.personLogo = action.payload.url;
			newState.personLogoId = action.payload.id;
			return newState;
		case "REMOVE_PERSON_LOGO":
			newState = handleStepEdit( newState, 2 );
			newState.personLogo = "";
			newState.personLogoId = "";
			return newState;
		case "SET_PERSON":
			newState = handleStepEdit( newState, 2 );
			newState.personId = action.payload.value;
			newState.personName = action.payload.label;
			return newState;
		case "SET_CAN_EDIT_USER":
			newState = handleStepEdit( newState, 2 );
			newState.canEditUser = ( action.payload === true ) ? 1 : 0;
			return newState;
		case "CHANGE_SOCIAL_PROFILE":
			newState = handleStepEdit( newState, 3 );
			newState.socialProfiles[ action.payload.socialMedium ] = action.payload.value;
			newState.errorFields = newState.errorFields.filter( errorField => {
				if ( action.payload.socialMedium === "facebookUrl" ) {
					return errorField !== "facebook_site";
				} else if ( action.payload.socialMedium === "twitterUsername" ) {
					return errorField !== "twitter_site";
				}
				return true;
			} );
			return newState;
		case "CHANGE_OTHERS_SOCIAL_PROFILE":
			newState = handleStepEdit( newState, 3 );
			newState.socialProfiles.otherSocialUrls[ action.payload.index ] = action.payload.value;
			newState.errorFields = newState.errorFields.filter( errorField => errorField !== `other_social_urls-${ action.payload.index }` );
			return newState;
		case "ADD_OTHERS_SOCIAL_PROFILE":
			newState = handleStepEdit( newState, 3 );
			newState.socialProfiles.otherSocialUrls = [ ...newState.socialProfiles.otherSocialUrls, action.payload.value ];
			return newState;
		case "REMOVE_OTHERS_SOCIAL_PROFILE":
			newState = handleStepEdit( newState, 3 );
			newState.socialProfiles.otherSocialUrls.splice( action.payload.index, 1 );
			newState.errorFields = removeFieldAndUpdateErrors( newState.errorFields, action.payload.index );
			return newState;
		case "SET_ERROR_FIELDS":
			newState.errorFields = action.payload;
			return newState;
		case "SET_STEP_ERROR":
			newState.stepErrors[ action.payload.step ] = action.payload.message;
			return newState;
		case "REMOVE_STEP_ERROR":
			newState.stepErrors = pickBy( newState.stepErrors, ( _value, key ) => key !== action.payload );
			return newState;
		case "SET_TRACKING":
			newState = handleStepEdit( newState, 4 );
			newState.tracking = action.payload;
			return newState;
		default:
			return newState;
	}
}
/* eslint-enable complexity */
