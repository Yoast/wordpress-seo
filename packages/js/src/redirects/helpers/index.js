export * from "./submit";
export * from "./validation";
