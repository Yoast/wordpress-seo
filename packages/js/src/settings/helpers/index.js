export * from "./i18n";
export * from "./search";
export * from "./submit";
export * from "./validation";
