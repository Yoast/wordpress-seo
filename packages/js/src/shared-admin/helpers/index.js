export { fixWordPressMenuScrolling } from "./fix-wordpress-menu-scrolling";
export { updateNotificationsCount } from "./notifications-count";
