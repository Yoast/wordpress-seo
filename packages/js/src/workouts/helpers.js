import apiFetch from "@wordpress/api-fetch";
import { colors } from "@yoast/style-guide";

/**
 * Unescapes HTML entities from a HTML string.
 *
 * @param {string} text The string to unescape
 * @returns {string} The escaped string.
 */
export function unescape( text ) {
	return jQuery( "<div>" ).html( text ).text();
}

/**
 * Retrieves the icons and colors for the icon for a certain result.
 *
 * @param {string} score The score for which to return the icon and color.
 *
 * @returns {Object} The icon and color for the score.
 */
export function getIconForScore( score ) {
	const icon = { icon: "circle", color: colors.$color_grey_disabled };

	switch ( score ) {
		case "loading":
			icon.color = colors.$color_green_medium_light;
			break;
		case "good":
			icon.color = colors.$color_green_medium;
			break;
		case "ok":
			icon.color = colors.$color_ok;
			break;
		case "bad":
			icon.color = colors.$color_red;
			break;
	}

	return icon;
}

/**
 * Searches WP for content.
 *
 * @param {string} term The search term.
 * @param {function} setSearchResult Sets the searchresults.
 * @returns {boolean} If the action was successful.
 */
export async function search( term, setSearchResult ) {
	try {
		const response = await apiFetch( {
			path: "wp/v2/search/?search=" + term,
		} );
		let searchResult = await response;
		searchResult = searchResult.map(
			( x ) => {
				return {
					value: x.id,
					label: x.title + " (" + x.url + ")",
					type: x.type,
				};
			}
		);
		setSearchResult( searchResult );
		return searchResult;
	} catch ( e ) {
		// URL() constructor throws a TypeError exception if url is malformed.
		setSearchResult( [] );
		console.error( e.message );
		return false;
	}
}
