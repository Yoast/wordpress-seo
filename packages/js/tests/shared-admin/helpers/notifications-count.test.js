/* eslint-disable react/prop-types */
import { describe, expect, it } from "@jest/globals";
import { updateNotificationsCount } from "../../../src/shared-admin/helpers";
import { render } from "../../test-utils";

/**
 * Simplified version of the admin bar count. With incorrect singular/plural.
 * @param {number} total The total number of notifications.
 * @returns {JSX.Element} The admin menu count.
 */
const AdminMenuCountHtml = ( { total } ) => (
	<span className={ `update-plugins count-${ total }` }>
		<span className="plugin-count" aria-hidden="true">{ total }</span>
		<span className="screen-reader-text">{ total } notifications</span>
	</span>
);

/**
 * Creates a fake admin menu.
 * The content is a simplified copy of the admin menu structure from the WordPress admin.
 * But with only the Yoast SEO list item in the menu.
 * @param {number} total The total number of notifications.
 * @returns {JSX.Element} The element.
 */
const FakeAdminMenu = ( { total = 2 } ) => (
	<div id="adminmenuwrap">
		<ul id="adminmenu">
			<li id="toplevel_page_wpseo_dashboard">
				<a href="admin.php?page=wpseo_dashboard">
					<div className="wp-menu-name">
						Yoast SEO <AdminMenuCountHtml total={ total } />
					</div>
				</a>
				<ul className="wp-submenu wp-submenu-wrap">
					<li className="wp-submenu-head" aria-hidden="true">
						Yoast SEO <AdminMenuCountHtml total={ total } />
					</li>
				</ul>
			</li>
		</ul>
	</div>
);

/**
 * Simplified version of the admin bar count. With incorrect singular/plural.
 * @param {number} total The total number of notifications.
 * @returns {JSX.Element} The admin bar count.
 */
const AdminBarCountHtml = ( { total } ) => (
	<div className={ `wp-core-ui wp-ui-notification yoast-issue-counter${ total === 0 ? " wpseo-no-adminbar-notifications" : "" }` }>
		<span className="yoast-issues-count" aria-hidden="true">${ total }</span>
		<span className="screen-reader-text">${ total } notifications</span>
	</div>
);

/**
 * Creates a fake admin bar.
 * The content is a simplified copy of the admin bar structure from the WordPress admin.
 * But with only the Yoast SEO section in the first menu.
 * @param {number} total The total number of notifications.
 * @returns {JSX.Element} The element.
 */
const FakeAdminBar = ( { total = 2 } ) => (
	<div id="wpadminbar" className="no-js">
		<div className="quicklinks" id="wp-toolbar" role="navigation" aria-label="Toolbar">
			<ul role="menu" id="wp-admin-bar-root-default" className="ab-top-menu">
				<li role="group" id="wp-admin-bar-wpseo-menu" className="menupop">
					<a className="ab-item" role="menuitem" aria-expanded="false" href="/wp-admin/admin.php?page=wpseo_dashboard">
						<AdminBarCountHtml total={ total } />
					</a>
					<div className="ab-sub-wrapper">
						<ul role="menu" id="wp-admin-bar-wpseo-menu-default" className="ab-submenu">
							<li role="group" id="wp-admin-bar-wpseo-notifications">
								<a className="ab-item" role="menuitem" href="/wp-admin/admin.php?page=wpseo_dashboard">
									Notifications <AdminBarCountHtml total={ total } />
								</a>
							</li>
						</ul>
					</div>
				</li>
			</ul>
		</div>
	</div>
);

describe( "updateNotificationsCount", () => {
	it( "does not throw an error when the elements are not there", () => {
		expect( updateNotificationsCount ).not.toThrow();
	} );

	it( "does not throw an error when the inner elements are not there", () => {
		render( <>
			<div id="toplevel_page_wpseo_dashboard">
				<div className="update-plugins" />
			</div>
			<div id="wp-admin-bar-wpseo-menu">
				<div className="yoast-issue-counter" />
			</div>
		</> );
		expect( updateNotificationsCount ).not.toThrow();
	} );

	test.each( [
		[ 3, "screen-reader plural in English" ],
		[ 1, "screen-reader singular in English" ],
		[ 0, "count-0 class triggering hidden css" ],
	] )( "updates the total in the admin menu to %i [%s]", ( total ) => {
		const { container } = render( <FakeAdminMenu /> );
		updateNotificationsCount( total );
		expect( container ).toMatchSnapshot();
	} );

	test.each( [
		[ 3, "screen-reader plural in English" ],
		[ 1, "screen-reader singular in English" ],
		[ 0, "wpseo-no-adminbar-notifications class triggering hidden css" ],
	] )( "updates the total in the admin bar to %i [%s]", ( total ) => {
		const { container } = render( <FakeAdminBar /> );
		updateNotificationsCount( total );
		expect( container ).toMatchSnapshot();
	} );
} );
