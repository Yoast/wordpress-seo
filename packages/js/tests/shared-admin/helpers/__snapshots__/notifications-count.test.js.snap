// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateNotificationsCount updates the total in the admin bar to 0 [wpseo-no-adminbar-notifications class triggering hidden css] 1`] = `
<div>
  <div
    class="no-js"
    id="wpadminbar"
  >
    <div
      aria-label="Toolbar"
      class="quicklinks"
      id="wp-toolbar"
      role="navigation"
    >
      <ul
        class="ab-top-menu"
        id="wp-admin-bar-root-default"
        role="menu"
      >
        <li
          class="menupop"
          id="wp-admin-bar-wpseo-menu"
          role="group"
        >
          <a
            aria-expanded="false"
            class="ab-item"
            href="/wp-admin/admin.php?page=wpseo_dashboard"
            role="menuitem"
          >
            <div
              class="wp-core-ui wp-ui-notification yoast-issue-counter wpseo-no-adminbar-notifications"
            >
              <span
                aria-hidden="true"
                class="yoast-issues-count"
              >
                0
              </span>
              <span
                class="screen-reader-text"
              >
                0 notifications
              </span>
            </div>
          </a>
          <div
            class="ab-sub-wrapper"
          >
            <ul
              class="ab-submenu"
              id="wp-admin-bar-wpseo-menu-default"
              role="menu"
            >
              <li
                id="wp-admin-bar-wpseo-notifications"
                role="group"
              >
                <a
                  class="ab-item"
                  href="/wp-admin/admin.php?page=wpseo_dashboard"
                  role="menuitem"
                >
                  Notifications 
                  <div
                    class="wp-core-ui wp-ui-notification yoast-issue-counter wpseo-no-adminbar-notifications"
                  >
                    <span
                      aria-hidden="true"
                      class="yoast-issues-count"
                    >
                      0
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      0 notifications
                    </span>
                  </div>
                </a>
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`updateNotificationsCount updates the total in the admin bar to 1 [screen-reader singular in English] 1`] = `
<div>
  <div
    class="no-js"
    id="wpadminbar"
  >
    <div
      aria-label="Toolbar"
      class="quicklinks"
      id="wp-toolbar"
      role="navigation"
    >
      <ul
        class="ab-top-menu"
        id="wp-admin-bar-root-default"
        role="menu"
      >
        <li
          class="menupop"
          id="wp-admin-bar-wpseo-menu"
          role="group"
        >
          <a
            aria-expanded="false"
            class="ab-item"
            href="/wp-admin/admin.php?page=wpseo_dashboard"
            role="menuitem"
          >
            <div
              class="wp-core-ui wp-ui-notification yoast-issue-counter"
            >
              <span
                aria-hidden="true"
                class="yoast-issues-count"
              >
                1
              </span>
              <span
                class="screen-reader-text"
              >
                1 notification
              </span>
            </div>
          </a>
          <div
            class="ab-sub-wrapper"
          >
            <ul
              class="ab-submenu"
              id="wp-admin-bar-wpseo-menu-default"
              role="menu"
            >
              <li
                id="wp-admin-bar-wpseo-notifications"
                role="group"
              >
                <a
                  class="ab-item"
                  href="/wp-admin/admin.php?page=wpseo_dashboard"
                  role="menuitem"
                >
                  Notifications 
                  <div
                    class="wp-core-ui wp-ui-notification yoast-issue-counter"
                  >
                    <span
                      aria-hidden="true"
                      class="yoast-issues-count"
                    >
                      1
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      1 notification
                    </span>
                  </div>
                </a>
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`updateNotificationsCount updates the total in the admin bar to 3 [screen-reader plural in English] 1`] = `
<div>
  <div
    class="no-js"
    id="wpadminbar"
  >
    <div
      aria-label="Toolbar"
      class="quicklinks"
      id="wp-toolbar"
      role="navigation"
    >
      <ul
        class="ab-top-menu"
        id="wp-admin-bar-root-default"
        role="menu"
      >
        <li
          class="menupop"
          id="wp-admin-bar-wpseo-menu"
          role="group"
        >
          <a
            aria-expanded="false"
            class="ab-item"
            href="/wp-admin/admin.php?page=wpseo_dashboard"
            role="menuitem"
          >
            <div
              class="wp-core-ui wp-ui-notification yoast-issue-counter"
            >
              <span
                aria-hidden="true"
                class="yoast-issues-count"
              >
                3
              </span>
              <span
                class="screen-reader-text"
              >
                3 notifications
              </span>
            </div>
          </a>
          <div
            class="ab-sub-wrapper"
          >
            <ul
              class="ab-submenu"
              id="wp-admin-bar-wpseo-menu-default"
              role="menu"
            >
              <li
                id="wp-admin-bar-wpseo-notifications"
                role="group"
              >
                <a
                  class="ab-item"
                  href="/wp-admin/admin.php?page=wpseo_dashboard"
                  role="menuitem"
                >
                  Notifications 
                  <div
                    class="wp-core-ui wp-ui-notification yoast-issue-counter"
                  >
                    <span
                      aria-hidden="true"
                      class="yoast-issues-count"
                    >
                      3
                    </span>
                    <span
                      class="screen-reader-text"
                    >
                      3 notifications
                    </span>
                  </div>
                </a>
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`updateNotificationsCount updates the total in the admin menu to 0 [count-0 class triggering hidden css] 1`] = `
<div>
  <div
    id="adminmenuwrap"
  >
    <ul
      id="adminmenu"
    >
      <li
        id="toplevel_page_wpseo_dashboard"
      >
        <a
          href="admin.php?page=wpseo_dashboard"
        >
          <div
            class="wp-menu-name"
          >
            Yoast SEO 
            <span
              class="update-plugins count-0"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                0
              </span>
              <span
                class="screen-reader-text"
              >
                0 notifications
              </span>
            </span>
          </div>
        </a>
        <ul
          class="wp-submenu wp-submenu-wrap"
        >
          <li
            aria-hidden="true"
            class="wp-submenu-head"
          >
            Yoast SEO 
            <span
              class="update-plugins count-0"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                0
              </span>
              <span
                class="screen-reader-text"
              >
                0 notifications
              </span>
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`updateNotificationsCount updates the total in the admin menu to 1 [screen-reader singular in English] 1`] = `
<div>
  <div
    id="adminmenuwrap"
  >
    <ul
      id="adminmenu"
    >
      <li
        id="toplevel_page_wpseo_dashboard"
      >
        <a
          href="admin.php?page=wpseo_dashboard"
        >
          <div
            class="wp-menu-name"
          >
            Yoast SEO 
            <span
              class="update-plugins count-1"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                1
              </span>
              <span
                class="screen-reader-text"
              >
                1 notification
              </span>
            </span>
          </div>
        </a>
        <ul
          class="wp-submenu wp-submenu-wrap"
        >
          <li
            aria-hidden="true"
            class="wp-submenu-head"
          >
            Yoast SEO 
            <span
              class="update-plugins count-1"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                1
              </span>
              <span
                class="screen-reader-text"
              >
                1 notification
              </span>
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`updateNotificationsCount updates the total in the admin menu to 3 [screen-reader plural in English] 1`] = `
<div>
  <div
    id="adminmenuwrap"
  >
    <ul
      id="adminmenu"
    >
      <li
        id="toplevel_page_wpseo_dashboard"
      >
        <a
          href="admin.php?page=wpseo_dashboard"
        >
          <div
            class="wp-menu-name"
          >
            Yoast SEO 
            <span
              class="update-plugins count-3"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                3
              </span>
              <span
                class="screen-reader-text"
              >
                3 notifications
              </span>
            </span>
          </div>
        </a>
        <ul
          class="wp-submenu wp-submenu-wrap"
        >
          <li
            aria-hidden="true"
            class="wp-submenu-head"
          >
            Yoast SEO 
            <span
              class="update-plugins count-3"
            >
              <span
                aria-hidden="true"
                class="plugin-count"
              >
                3
              </span>
              <span
                class="screen-reader-text"
              >
                3 notifications
              </span>
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
